scipy/optimize/_sda.py

```diff
@@ -0,0 +1,589 @@
+    approach generalizes CSA [2]_ (Classical Simulated Annealing) and FSA (Fast
+    Simulated Annealing) to find the neighborhood of minima and introduces an
+    additional annealing process for the best solution found.
+
+    This algorithm uses a distorted Cauchy-Lorentz visiting distribution, with
+    its shape controlled by the parameter :math:`q_{v}`
+
+    .. math::
+
+        g_{q_{v}}(\\Delta x(t)) \\propto \\frac{ \\
+        \\left[T_{q_{v}}(t) \\right]^{-\\frac{D}{3-q_{v}}}}{ \\
+        \\left[{1+(q_{v}-1)\\frac{(\Delta x(t))^{2}} { \\
+        \\left[T_{q_{v}}(t)\\right]^{\\frac{2}{3-q_{v}}}}}\\right]^{ \\
+        \\frac{1}{q_{v}-1}+\\frac{D-1}{2}}}
+
+    Where :math:`t` is the artificial time. This visiting distribution is used
+    to generate a trial jump distance :math:`\Delta x(t)` of variable
+    :math:`x(t)` under artificial temperature :math:`T_{q_{v}}(t)`.
+
+    From the starting point, after calling the visiting distribution
+    function, the acceptance probability is computed as follows:
+
+    .. math::
+
+        p_{q_{a}} = \min{\{1,\\left[1-(1-q_{a}) \\beta \\Delta E \\right]^{ \\
+        \\frac{1}{1-q_{a}}}\\}}
+
+    Where :math:`q_{a}` is a acceptance parameter. For :math:`q_{a}<1`, zero
+    acceptance probability is assigned to the cases where
+
+    .. math::
+
+        [1-(1-q_{a}) \\beta \\Delta E] < 0
+
+    The artificial temperature :math:`T_{q_{v}}(t)` is decreased according to
+
+    .. math::
+
+        T_{q_{v}}(t) = T_{q_{v}}(1) \\frac{2^{q_{v}-1}-1}{\\left( \\
+        1 + t\\right)^{q_{v}-1}-1}
+
+    Where :math:`q_{v}` is the visiting parameter.
+
+    .. versionadded:: 0.19.0
+
+    References
+    ----------
+    .. [1] Tsallis C (1988). "Possible generalization of Boltzmann-Gibbs
+        statistics." Journal of Statistical Physics, 52, 479-487.
+    .. [2] Tsallis C, Stariolo DA (1996). "Generalized Simulated Annealing."
+        Physica A, 233, 395-406.
+    .. [3] Xiang Y, Sun DY, Fan W, Gong XG (1997). "Generalized Simulated
+        Annealing Algorithm and Its Application to the Thomson Model."
+        Physics Letters A, 233, 216-220.
+    .. [4] Xiang Y, Gong XG (2000a). "Efficiency of Generalized Simulated
+        Annealing." PHYSICAL REVIEW E, 62, 4473.
+    .. [5] Xiang Y, Gubian S, Suomela B, Hoeng (2013). "Generalized Simulated
+        Annealing for Efficient Global Optimization: the GenSA Package for
+        R". The R Journal, Volume 5/1, June 2013.
+        http://journal.r-project.org/.
+    .. [6] Mullen, K. (2014). Continuous Global Optimization in R. Journal of
+        Statistical Software, 60(6), 1 - 45.
+        http://dx.doi.org/10.18637/jss.v060.i06
+
+    Examples
+    --------
+    The following example is a 10-dimensional problem, with many local minima.
+    The function involved is called Rastrigin
+    (https://en.wikipedia.org/wiki/Rastrigin_function)
+    >>> import numpy as np
```

Comment:
This import should be removed, it's implicit in all our examples. Please replace with a blank line, then the html gets rendered correctly (always blank lines around `>>>`)

Reply:
Sure ! Thanks for this.